### ISSUE TITLE
Restrict permissions in approver policy

### ISSUE THREAD

**Issue by randazzop** [aws-samples/iam-identity-center-team#81]
Currently, the approver policy configuration allows a user group to be an approver for an entire account or organization unit. 

This is a too wide configuration. It would be necessary to modify the approver policy configuration by adding a list of permission sets or alternatively a list of eligibility policies, in order to have more granular control.

**Comment by tawoyinfa** [comment]
@randazzop good observation. We will add add this as a feature request and include in the roadmap.

**Comment by matthewcbaker** [comment]
I am evaluating TEAM right now, and it looks fantastic so far, but I came looking to see if someone else had noted this issue as I can also see it being a problem.

An example scenario that demonstrates this:
- Developers can get ReadOnly, Development and Admin access to Release account.
- Peer approval is required for ReadOnly and Development access.
- IT approval is required for Admin access.

I am glad that this has already been seen by @tawoyinfa as a valid feature request.

**Comment by matthewcbaker** [comment]
It's unfortunate that this has been automatically closed after such a short time - it's still the main issue holding back deployment of TEAM.

**Comment by matthewcbaker** [comment]
@tawoyinfa, this also doesn't appear to be in the roadmap as you [mentioned above](https://github.com/aws-samples/iam-identity-center-team/issues/81#issuecomment-1727503793).

Do you think that having different groups of approvers for different roles within the same account is still a valid request?

**Comment by matthewcbaker** [comment]
Hi @tawoyinfa,

I see that you have re-opened #39, regarding the eligibility policies, as it is still seen as important.  As I noted there that, whilst difficult to maintain, it can be addressed by creating multiple groups with the same users in.

Unfortunately this issue cannot be addressed in the same way.

It is still important to us to be able to allocate 'Admin' role approval for an account to one set of users (our Global Admins), and 'General' role approval for the same account to another (our Account Managers).  This is not currently possible.

Effectively, this request is for the same set of permissions relationships that is in IAM Identity Center itself - a list of Account-Role-Group mappings.

Could you re-open this issue too please, to show that it is still of interest.

Thank you.

**Comment by matthewcbaker** [comment]
Thank you @tawoyinfa, much appreciated.

**Comment by matthewcbaker** [comment]
This is still an issue for us

**Comment by matthewcbaker** [comment]
Out of all the issues, this one is still the biggest, as we have several different types of situations in play right now that effectively can be represented by the scenario I described above in https://github.com/aws-samples/iam-identity-center-team/issues/81#issuecomment-1772343987, where two different groups of approvers are needed within the same account.

**Comment by matthewcbaker** [comment]
Still a big priority

**Comment by cnclickup** [comment]
Agreed, the ability to scope approval patterns within the same accounts/OUs to specific eligibility policies or permission sets is key for us to be able to effectively role out permission matrix support.

**Comment by danluera** [comment]
I actually dissent on this one.  But, then again maybe the way our Organization is set up has something to do with that.  We have a division, product, environment hierarchy to our OUs.  So, we have definite "account owners" who are ultimately responsible for the accounts.  Which, given that kind of setup allows the current Approvers implementation to make sense.

Giving ReadOnly (by this I mean something more pared down than the AWS managed Read EVERYTHING policy) to folks shouldn't even really need approval in my opinion.  You're on Team A, which support Product A?  Yeah, look around.  You can't change anything or exfiltrate the data.  Now you want to change something?  Ok, we'll need to get approval for that.

Actually, our goal is ReadOnly for product support teams in their product accounts as a baseline.  TEAM is going to be purely intended to temporarily elevate them for various types of admin access.

Auditing who has Approval authority in an account becomes much simpler as well.

**Comment by cnclickup** [comment]
I agree the organization setup you outlined will have some mitigations for this issue. However, we have have patterns where there are owners of accounts who have different scopes. For example: Even if the account is scoped down to a specific service and not shared, we want to use TEAM to grant permission for things like marketplace subscriptions permission sets. In that case there are finance/legal owners as approvers specific to those permissions - vs approval for things like modify access that would be scoped to the owning squad.

**Comment by matthewcbaker** [comment]
@danluera, we have definite owners for all our accounts too, but as @cnclickup says - there are still services within those accounts owned by IT, Finance, Legal, and other teams that apply across the entire business.  At the moment then pre-authorising those groups (so there's a record of why something has happened) is the best we can do.

Within a single team then there are also day-to-day activities that a team needs to do within their own account, but occasionally then they need to do A or B.  For activity A we want their own team to authorise them, but for activity B then we want someone higher up in the business to do it (like IT, Finance or Legal).  The main alternatives here are to a) let their own team authorise both, so there's at least /someone/ authorising them; b) having all groups of users as authorisers, and having people ignore the ones that aren't for them; c) pre-authorising everything so there's just a record.

**Comment by matthewcbaker** [comment]
This is still a priority - all previous reasons are still valid.

**Comment by mbaker-itrs** [comment]
Still a priority.  As @cnclickup says - this is required for anyone with more than a very basic matrix/RBAC in place.

**Comment by matthewcbaker** [comment]
This is still one of the biggest issues

**Comment by mbaker-itrs** [comment]
This is still the biggest TEAM issue

**Comment by matthewcbaker** [comment]
Still an issue

**Comment by matthewcbaker** [comment]
Still the biggest issue

**Comment by mbaker-itrs** [comment]
This is still a key issue

**Comment by m-barthelemy** [comment]
This would still be important for us to have

**Comment by m-barthelemy** [comment]
still needed

**Comment by AlDawoode** [comment]
I believe this feature is really needed by everyone using the TEAM solution today.
We would like to be able to create multiple eligibility policies for one account, something like:


Id | Name | Type | Ticket no | PermissionSet (new feature) | Approver groups
-- | -- | -- | -- | -- | --
1111111111 | my-account-pr | Account | ICET-1111 | DeveloperRole | my-account-pr-developer-approval
1111111111 | my-account-pr | Account | ICET-222 | AuditRole | my-account-pr-audit-approval
